File: java/TowerOfHanoi.java
import java.util.Scanner;

public class TowerOfHanoi {
    public static void TOH(int n, String src, String helper, String dest) {
        if(n == 1){
            System.out.println("Disk " + n + " Transferred: from " + src + " to " + dest);
            return;
        }
        TOH(n-1, src, dest, helper);

        System.out.println("Disk " + n + " Transferred: from " + src + " to " + dest);

        TOH(n-1, helper, src, dest);
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.close();
        
        TOH(n, "Source", "Helper", "Destination");
    }
}

File: CPP/towerOfHanoi.cpp
#include <iostream>
using namespace std;

void ToH(int n, char source, char aux, char dest){
    if (n==1){
        cout << "Moving disc 1 from " << source << " to " << dest << endl; 
        return;
    }
    ToH(n-1, source, dest, aux);
    cout << "Moving disc " << n << " from " << source << " to " << dest << endl;
    ToH(n-1, source, aux, dest);
}

int main(){
    int n;
    cout << "Enter no of discs: ";
    cin >> n;

    ToH(n, 'A', 'B', 'C');
    
    return 0;
}